MantidQt/CustomInterfaces/src/Indirect/Quasi.cpp

```diff
@@ -410,5 +413,69 @@ void Quasi::previewSpecChanged(int value) {
   updateMiniPlot();
 }
 
+/**
+ * Handles saving the workspace when save is clicked
+ */
+void Quasi::saveClicked() {
+  QString saveDirectory = QString::fromStdString(
+      Mantid::Kernel::ConfigService::Instance().getString(
+          "defaultsave.directory"));
+  const auto fitWS = m_QuasiAlg->getPropertyValue("OutputWorkspaceFit");
+  QString QfitWS = QString::fromStdString(fitWS);
+  const auto fitPath = saveDirectory + QfitWS + ".nxs";
+  addSaveWorkspaceToQueue(QfitWS, fitPath);
+
+  const auto resultWS = m_QuasiAlg->getPropertyValue("OutputWorkspaceResult");
+  QString QresultWS = QString::fromStdString(resultWS);
+  const auto resultPath = saveDirectory + QresultWS + ".nxs";
+  addSaveWorkspaceToQueue(QresultWS, resultPath);
+  m_batchAlgoRunner->executeBatchAsync();
+}
+
+/**
+ * Handles plotting the selected plot when plot is clicked
+ */
+void Quasi::plotClicked() {
+  // Output options
+  std::string plot = m_uiForm.cbPlot->currentText().toStdString();
+  const auto resultName = m_QuasiAlg->getPropertyValue("OutputWorkspaceResult");
+  if (plot == "Prob" || plot == "All") {
+    const auto probWS = m_QuasiAlg->getPropertyValue("OutputWorkspaceProb");
+    QString QprobWS = QString::fromStdString(probWS);
+    IndirectTab::plotSpectrum(QprobWS, 1, 2);
+  }
+  if (plot == "Fit" || plot == "All") {
+    std::string fitName = m_QuasiAlg->getPropertyValue("OutputWorkspaceFit");
+    fitName.pop_back();
+    fitName.append("_0");
+    QString QfitWS = QString::fromStdString(fitName);
+    IndirectTab::plotSpectra(QfitWS, {0, 1, 2, 4});
+  }
+
+  MatrixWorkspace_sptr resultWS =
+      AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>(resultName);
+  int numSpectra = (int)resultWS->getNumberHistograms();
+
+  QString QresultWS = QString::fromStdString(resultName);
+  auto paramNames = {"Amplitude", "FWHM", "Beta"};
+  for (std::string paramName : paramNames) {
+
+    if (plot == paramName || plot == "All") {
+      std::vector<int> spectraIndices = {};
+      for (int i = 0; i <= numSpectra; i++) {
+        auto axisLabel = resultWS->getAxis(1)->label(i);
+
+        auto found = axisLabel.find(paramName);
+        if (found != std::string::npos) {
+          spectraIndices.push_back(i);
+          if (spectraIndices.size() == 3) {
```

Comment:
Changing the program to `Stretched Exponential` means that only one curve is plotted rather than 3. The above if statement stops any plots being produced when the Program is `Stretch Exponential`. There should possibly be a special case for this?